<launch>

  <arg name="output" default="screen" />
  <arg name="respawn" default="false" />
  <arg name="visualize" default="false" />

  <arg name="node_name" default="video_recorder" />
  <arg name="image_topic" default="image_raw" />
  <arg name="reset_service" default="reset_recorder" />
  <arg name="auto_reset" default="true" />

  <arg name="target" default="output.mp4" />
  <arg name="format" default="X264" />
  <arg name="width" default="1920" />
  <arg name="height" default="1080" />
  <arg name="fps" default="30" />

  <arg name="pipeline" default="appsrc ! video/x-raw, format=(string)BGR ! videoconvert ! video/x-raw, format=BGRx ! nvvidconv ! video/x-raw(memory:NVMM), format=(string)NV12 ! omxh264enc ! qtmux ! filesink location=$(arg target)" />

  <node pkg="video_recorder" type="video_recorder_node" name="$(arg node_name)" 
        output="$(arg output)" respawn="$(arg respawn)"> 
    <param name="backend" type="str" value="gstreamer" />
    <param name="target" type="str" value="$(arg pipeline)" />
    <param name="format" type="str" value="$(arg format)" />
    <param name="width" type="int" value="$(arg width)" />
    <param name="height" type="int" value="$(arg height)" />
    <param name="fps" type="double" value="$(arg fps)" />
    <param name="auto_reset" type="bool" value="$(arg auto_reset)" />
    <remap from="reset" to="$(arg reset_service)" />
    <remap from="image" to="$(arg image_topic)" />
  </node>

  <node pkg="image_view" type="image_view" name="$(arg node_name)_view" 
        output="$(arg output)" respawn="$(arg respawn)" if="$(arg visualize)">
    <remap from="image" to="$(arg image_topic)" />
  </node>

</launch>
